--- src/Terraria/Terraria/GameContent/Events/CultistRitual.cs
+++ src/Avalon/Terraria/GameContent/Events/CultistRitual.cs
@@ -47,7 +_,7 @@
 		}
 
 		private static bool CheckRitual(int x, int y) {
-			if (delay != 0 || !Main.hardMode || !NPC.downedGolemBoss || !NPC.downedBoss3)
+			if (delay != 0 || !Main.hardMode || !NPC.downedGolemBoss || !NPC.downedBoss3 || !NPC.downedGolemSinceCultistsKilled)
 				return false;
 
 			if (y < 7 || WorldGen.SolidTile(Main.tile[x, y - 7]))
