--- src/Terraria/Terraria/GameContent/Creative/CreativePowers.cs
+++ src/Avalon/Terraria/GameContent/Creative/CreativePowers.cs
@@ -862,7 +_,7 @@
 				uIPanel.Append(uIVerticalSlider);
 				uIPanel.OnUpdate += CreativePowersHelper.UpdateUseMouseInterface;
 				uIVerticalSlider.OnUpdate += UpdateSliderColorAndShowMultiplierMouseOver;
-				AddIndication(uIPanel, 0f, "x3", "Images/UI/WorldCreation/IconDifficultyMaster", MouseOver_Master);
+				//AddIndication(uIPanel, 0f, "x3", "Images/UI/WorldCreation/IconDifficultyMaster", MouseOver_Master);
 				AddIndication(uIPanel, 0.33333334f, "x2", "Images/UI/WorldCreation/IconDifficultyExpert", MouseOver_Expert);
 				AddIndication(uIPanel, 2f / 3f, "x1", "Images/UI/WorldCreation/IconDifficultyNormal", MouseOver_Normal);
 				AddIndication(uIPanel, 1f, "x0.5", "Images/UI/WorldCreation/IconDifficultyCreative", MouseOver_Journey);
@@ -922,7 +_,7 @@
 				UIVerticalSlider uIVerticalSlider = affectedElement as UIVerticalSlider;
 				if (uIVerticalSlider != null) {
 					uIVerticalSlider.EmptyColor = Color.Black;
-					Color color = uIVerticalSlider.FilledColor = (Main.masterMode ? Main.hcColor : (Main.expertMode ? Main.mcColor : ((!(StrengthMultiplierToGiveNPCs < 1f)) ? Color.White : Main.creativeModeColor)));
+					Color color = uIVerticalSlider.FilledColor = Main.expertMode ? Main.mcColor : ((!(StrengthMultiplierToGiveNPCs < 1f)) ? Color.White : Main.creativeModeColor);
 				}
 			}
 
