--- src/Terraria/Terraria/GameContent/LootSimulation/LootSimulator.cs
+++ src/Avalon/Terraria/GameContent/LootSimulation/LootSimulator.cs
@@ -55,7 +_,7 @@
 			string str = "";
 			Stopwatch stopwatch = new Stopwatch();
 			stopwatch.Start();
-			for (int i = -65; i < 665; i++) {
+			for (int i = -65; i < Main.maxNPCTypes; i++) {
 				if (TryGettingLootFor(i, timesMultiplier, out string outputText))
 					str = str + outputText + "\n\n";
 			}
