--- src/Terraria/Terraria/GameContent/LootSimulation/LootSimulationItemCounter.cs
+++ src/Avalon/Terraria/GameContent/LootSimulation/LootSimulationItemCounter.cs
@@ -6,8 +_,8 @@
 {
 	public class LootSimulationItemCounter
 	{
-		private long[] _itemCountsObtained = new long[5045];
+		private long[] _itemCountsObtained = new long[Main.maxItemTypes];
-		private long[] _itemCountsObtainedExpert = new long[5045];
+		private long[] _itemCountsObtainedExpert = new long[Main.maxItemTypes];
 		private long _totalTimesAttempted;
 		private long _totalTimesAttemptedExpert;
 
