--- src/Terraria/Terraria/GameContent/UI/States/UIWorldCreation.cs
+++ src/Avalon/Terraria/GameContent/UI/States/UIWorldCreation.cs
@@ -311,25 +_,22 @@
 		}
 
 		private void AddWorldDifficultyOptions(UIElement container, float accumualtedHeight, MouseEvent clickEvent, string tagGroup, float usableWidthPercent) {
-			WorldDifficultyId[] array = new WorldDifficultyId[4] {
+			WorldDifficultyId[] array = new WorldDifficultyId[3] {
 				WorldDifficultyId.Creative,
 				WorldDifficultyId.Normal,
-				WorldDifficultyId.Expert,
+				WorldDifficultyId.Expert
-				WorldDifficultyId.Master
 			};
 
-			LocalizedText[] array2 = new LocalizedText[4] {
+			LocalizedText[] array2 = new LocalizedText[3] {
 				Language.GetText("UI.Creative"),
 				Language.GetText("UI.Normal"),
-				Language.GetText("UI.Expert"),
+				Language.GetText("UI.Expert")
-				Language.GetText("UI.Master")
 			};
 
-			LocalizedText[] array3 = new LocalizedText[4] {
+			LocalizedText[] array3 = new LocalizedText[3] {
 				Language.GetText("UI.WorldDescriptionCreative"),
 				Language.GetText("UI.WorldDescriptionNormal"),
-				Language.GetText("UI.WorldDescriptionExpert"),
+				Language.GetText("UI.WorldDescriptionExpert")
-				Language.GetText("UI.WorldDescriptionMaster")
 			};
 
 			Color[] array4 = new Color[4] {
@@ -648,9 +_,6 @@
 				case WorldDifficultyId.Expert:
 					Main.GameMode = 1;
 					break;
-				case WorldDifficultyId.Master:
-					Main.GameMode = 2;
-					break;
 			}
 
 			switch (_optionEvil) {
@@ -712,9 +_,6 @@
 						break;
 					case 2:
 						_optionDifficulty = WorldDifficultyId.Expert;
-						break;
-					case 3:
-						_optionDifficulty = WorldDifficultyId.Master;
 						break;
 					case 4:
 						_optionDifficulty = WorldDifficultyId.Creative;
