--- src/Terraria/Terraria/ID/DustID.cs
+++ src/Avalon/Terraria/ID/DustID.cs
@@ -126,7 +_,6 @@
 		public const short ArgonMoss = 301;
 		public const short Terragrim = 302;
 		public const short SteampunkSteam = 303;
-		public const short Count = 304;
 		public const short t_BorealWood = 214;
 		public const short t_Slime = 4;
 		public const short t_SteampunkMetal = 8;
@@ -144,11 +_,25 @@
 		public const short t_Flesh = 5;
 		public const short t_Frozen = 80;
 		public const short GemAmethyst = 86;
-		public const short GemTopaz = 87;
+		public const short GemTopaz = 87; 
 		public const short GemSapphire = 88;
 		public const short GemEmerald = 89;
 		public const short GemRuby = 90;
 		public const short GemDiamond = 91;
 		public const short GemAmber = 138;
+		public const short OpalBolt = 227;
+		public const short SoulofLight = 304;
+		public const short SoulofNight = 305;
+		public const short SoulofFright = 306;
+		public const short SoulofMight = 307;
+		public const short SoulofSight = 308;
+		public const short SoulofFlight = 309;
+		public const short SoulofBlight = 310;
+		public const short SouloftheJungle = 311;
+		public const short SoulofTorture = 312;
+		public const short ContagionSpray = 313;
+		public const short Berserker = 314;
+		public const short SoulofDelight = 315;
+		public const short Count = 316;
 	}
 }
