--- src/Terraria/Terraria/ID/TileID.cs
+++ src/Avalon/Terraria/ID/TileID.cs
@@ -2,7 +_,7 @@
 
 namespace Terraria.ID
 {
-	public class TileID
+	public partial class TileID
 	{
 		public static class Sets
 		{
@@ -823,6 +_,5 @@
 		public const ushort SliceOfCake = 621;
 		public const ushort TeaKettle = 622;
 		public const ushort PottedCrystalPlants = 623;
-		public const ushort Count = 624;
 	}
 }
