--- src/Terraria/Terraria/IO/ResourcePack.cs
+++ src/Avalon/Terraria/IO/ResourcePack.cs
@@ -84,9 +_,9 @@
 		public IContentSource GetContentSource() {
 			if (_needsReload) {
 				if (_isCompressed)
-					_contentSource = new ZipContentSource(FullPath, "Content");
+					_contentSource = new ZipContentSource(FullPath, "Avalon Content");
 				else
-					_contentSource = new FileSystemContentSource(Path.Combine(FullPath, "Content"));
+					_contentSource = new FileSystemContentSource(Path.Combine(FullPath, "Avalon Content"));
 
 				_contentSource.ContentValidator = VanillaContentValidator.Instance;
 				_needsReload = false;
