--- src/Terraria/Terraria/IO/WorldFile.cs
+++ src/Avalon/Terraria/IO/WorldFile.cs
@@ -386,6 +_,7 @@
 						WorldGen.WaterCheck();
 						WorldGen.gen = false;
 						NPC.setFireFlyChance();
+						Main.InitNPCLifeBytes();
 						if (Main.slimeRainTime > 0.0)
 							Main.StartSlimeRain(announce: false);
 
@@ -935,6 +_,7 @@
 			writer.Write(NPC.boughtBunny);
 			writer.Write(NPC.downedEmpressOfLight);
 			writer.Write(NPC.downedQueenSlime);
+			writer.Write(NPC.downedGolemSinceCultistsKilled);
 			return (int)writer.BaseStream.Position;
 		}
 
@@ -1633,10 +_,12 @@
 			if (versionNumber >= 223) {
 				NPC.downedEmpressOfLight = reader.ReadBoolean();
 				NPC.downedQueenSlime = reader.ReadBoolean();
+				NPC.downedGolemSinceCultistsKilled = reader.ReadBoolean();
 			}
 			else {
 				NPC.downedEmpressOfLight = false;
 				NPC.downedQueenSlime = false;
+				NPC.downedGolemSinceCultistsKilled = false;
 			}
 		}
 
