--- src/Terraria/Terraria/IO/WorldFile.cs
+++ src/Avalon/Terraria/IO/WorldFile.cs
@@ -883,8 +_,8 @@
 			writer.Write(NPC.savedGolfer);
 			writer.Write(Main.invasionSizeStart);
 			writer.Write(_tempCultistDelay);
-			writer.Write((short)665);
+			writer.Write((short)Main.maxNPCTypes);
-			for (int j = 0; j < 665; j++) {
+			for (int j = 0; j < Main.maxNPCTypes; j++) {
 				writer.Write(NPC.killCount[j]);
 			}
 
@@ -1489,7 +_,7 @@
 
 			int num2 = reader.ReadInt16();
 			for (int i = 0; i < num2; i++) {
-				if (i < 665)
+				if (i < Main.maxNPCTypes)
 					NPC.killCount[i] = reader.ReadInt32();
 				else
 					reader.ReadInt32();
@@ -1694,7 +_,7 @@
 
 					if ((b3 & 4) == 4) {
 						tile.wall = reader.ReadByte();
-						if (tile.wall >= 316)
+						if (tile.wall >= Main.maxWallTypes)
 							tile.wall = 0;
 
 						if ((b & 0x10) == 16)
@@ -1744,7 +_,7 @@
 						if ((b & 0x40) == 64) {
 							b4 = reader.ReadByte();
 							tile.wall = (ushort)((b4 << 8) | tile.wall);
-							if (tile.wall >= 316)
+							if (tile.wall >= Main.maxWallTypes)
 								tile.wall = 0;
 						}
 					}
@@ -2570,7 +_,7 @@
 
 					if (fileIO.ReadBoolean()) {
 						tile.wall = fileIO.ReadByte();
-						if (tile.wall >= 316)
+						if (tile.wall >= Main.maxWallTypes)
 							tile.wall = 0;
 
 						if (versionNumber >= 48 && fileIO.ReadBoolean())
