--- src/Terraria/Terraria/UI/ItemSlot.cs
+++ src/Avalon/Terraria/UI/ItemSlot.cs
@@ -1013,7 +_,7 @@
 							Recipe.FindRecipes();
 						}
 					}
-					else if (Main.mouseRight && inv[slot].type > 0 && inv[slot].type < 5088 && ItemID.Sets.IsFishingCrate[inv[slot].type]) {
+					else if (Main.mouseRight && inv[slot].type > 0 && inv[slot].type < Main.maxItemTypes && ItemID.Sets.IsFishingCrate[inv[slot].type]) {
 						if (Main.mouseRightRelease) {
 							player.OpenFishingCrate(inv[slot].type);
 							inv[slot].stack--;
@@ -2105,7 +_,7 @@
 		}
 
 		public static Color GetItemLight(ref Color currentColor, ref float scale, int type, bool outInTheWorld = false) {
-			if (type < 0 || type > 5088)
+			if (type < 0 || type > Main.maxItemTypes)
 				return currentColor;
 
 			if (type == 662 || type == 663) {
