--- src/Terraria/Terraria/UI/ItemSorting.cs
+++ src/Avalon/Terraria/UI/ItemSorting.cs
@@ -809,7 +_,7 @@
 			list.Add(ItemSortingLayers.LastTilesCommon);
 			list.Add(ItemSortingLayers.LastNotTrash);
 			list.Add(ItemSortingLayers.LastTrash);
-			for (int i = -48; i < 5088; i++) {
+			for (int i = -48; i < Main.maxItemTypes; i++) {
 				Item item = new Item();
 				item.netDefaults(i);
 				list2.Add(item);
