--- src/Terraria/Terraria/Chest.cs
+++ src/Avalon/Terraria/Chest.cs
@@ -13,9 +_,9 @@
 	public class Chest
 	{
 		public const float chestStackRange = 250f;
-		public const int maxChestTypes = 52;
+		public const int maxChestTypes = 59;
-		public static int[] chestTypeToIcon = new int[52];
+		public static int[] chestTypeToIcon = new int[maxChestTypes];
-		public static int[] chestItemSpawn = new int[52];
+		public static int[] chestItemSpawn = new int[maxChestTypes];
 		public const int maxChestTypes2 = 14;
 		public static int[] chestTypeToIcon2 = new int[14];
 		public static int[] chestItemSpawn2 = new int[14];
@@ -95,14 +_,14 @@
 			array2[33] = (array[33] = 2559);
 			array2[34] = (array[34] = 2574);
 			array2[35] = (array[35] = 2612);
-			array2[36] = 327;
-			array[36] = 2612;
+			array2[36] = 327; // TODO: 3433
+			array[36] = 2612; // TODO: 3433
 			array2[37] = (array[37] = 2613);
-			array2[38] = 327;
-			array[38] = 2613;
+			array2[38] = 327; // TODO: 3493
+			array[38] = 2613; // TODO: 3493
 			array2[39] = (array[39] = 2614);
-			array2[40] = 327;
-			array[40] = 2614;
+			array2[40] = 327; // TODO: 3522
+			array[40] = 2614; // TODO: 3522
 			array2[41] = (array[41] = 2615);
 			array2[42] = (array[42] = 2616);
 			array2[43] = (array[43] = 2617);
@@ -114,6 +_,16 @@
 			array2[49] = (array[49] = 3180);
 			array2[50] = (array[50] = 3125);
 			array2[51] = (array[51] = 3181);
+			array2[52] = (array[52] = 3104);
+			array2[53] = 3103;
+			array[53] = 3104;
+			array2[54] = (array[54] = 3128);
+			array2[55] = 3127;
+			array[55] = 3128;
+			array2[56] = (array[56] = 3393);
+			array2[57] = 3392;
+			array[57] = 3393;
+			array2[58] = (array[58] = 3409);
 			int[] array3 = chestItemSpawn2;
 			int[] array4 = chestTypeToIcon2;
 			array4[0] = (array3[0] = 3884);
@@ -794,7 +_,8 @@
 					num4 = 3099;
 
 				if (num4 != 0) {
+					// TODO: Main.travelShop[0] = 3188;
-					for (int k = 0; k < 40; k++) {
+					for (int k = 1; k < 40; k++) {
 						if (Main.travelShop[k] == num4) {
 							num4 = 0;
 							break;
@@ -1016,6 +_,8 @@
 					if (Main.player[Main.myPlayer].HasItem(1258)) {
 						array[num].SetDefaults(1261);
 						num++;
+
+						// TODO: if (Main.superHardMode { item[num].SetDefaults(3557); num++; }
 					}
 					if (Main.player[Main.myPlayer].HasItem(1835)) {
 						array[num].SetDefaults(1836);
@@ -1463,6 +_,7 @@
 					num++;
 					array[num].SetDefaults(849);
 					num++;
+					// TODO: array[num].SetDefaults(3633); num++;
 					array[num++].SetDefaults(3616);
 					array[num++].SetDefaults(2799);
 					array[num++].SetDefaults(3619);
@@ -1574,6 +_,19 @@
 						array[num].SetDefaults(781);
 						num++;
 					}
+					// TODO: This
+					/*
+						else if (Main.player[Main.myPlayer].zoneBooger)
+						{
+							item[num].SetDefaults(3388);
+							num++;
+						}
+						else if (Main.player[Main.myPlayer].zoneJungle)
+						{
+							item[num].SetDefaults(3043);
+							num++;
+						}
+					 */
 					else {
 						array[num].SetDefaults(780);
 						num++;
@@ -2480,6 +_,38 @@
 
 						break;
 					}
+
+				case 24:
+					// TODO: Iceman shop
+					/*item[num].SetDefaults(2161);
+					item[num].value = Item.buyPrice(0, 2);
+					num++;
+					item[num].SetDefaults(2306);
+					item[num].value = Item.buyPrice(0, 0, 25);
+					num++;
+					item[num].SetDefaults(2772);
+					item[num].value = Item.buyPrice(0, 4);
+					num++;
+					item[num].SetDefaults(3014);
+					num++;
+					item[num].SetDefaults(3016);
+					num++;
+					item[num].SetDefaults(3017);
+					item[num].value = Item.buyPrice(0, 10);
+					num++;
+					item[num].SetDefaults(974);
+					item[num].value = Item.buyPrice(0, 0, 0, 10);
+					num++;
+					if (NPC.oblivionDead) {
+						item[num].SetDefaults(3015);
+						item[num].value = Item.buyPrice(2);
+						num++;
+					}
+					if (Main.player[Main.myPlayer].zoneSnow) {
+						item[num].SetDefaults(2769);
+						num++;
+					}*/
+					break;
 			}
 
 			if (type != 19 && type != 20 && flag && !Main.player[Main.myPlayer].ZoneCorrupt && !Main.player[Main.myPlayer].ZoneCrimson) {
