--- src/Terraria/Terraria/Collision.cs
+++ src/Avalon/Terraria/Collision.cs
@@ -6,7 +_,7 @@
 
 namespace Terraria
 {
-	public class Collision
+	public partial class Collision
 	{
 		public static bool stair;
 		public static bool stairFall;
@@ -1832,7 +_,7 @@
 			Vector2 vector3 = default(Vector2);
 			for (int i = num; i < value2; i++) {
 				for (int j = value3; j < value4; j++) {
-					if (Main.tile[i, j] != null && Main.tile[i, j].liquid > 0 && Main.tile[i, j - 1].liquid == 0 && (!Main.tile[i, j].lava() || lavaWalk)) {
+					if (Main.tile[i, j] != null && Main.tile[i, j].liquid > 0 && Main.tile[i, j - 1].liquid == 0 && (!Main.tile[i, j].lava())) {
 						int num2 = (int)Main.tile[i, j].liquid / 32 * 2 + 2;
 						vector3.X = i * 16;
 						vector3.Y = j * 16 + 16 - num2;
@@ -1918,6 +_,15 @@
 						if (num8 == num6)
 							result.X = vector.X + 0.01f;
 					}
+					else if (vector3.Y >= vector4.Y + (float)num9 && !Main.tileSolidTop[Main.tile[i, j].type]) {
+						up = true;
+						num7 = i;
+						num8 = j;
+
+						if (num8 == num6) {
+							result.X = vector.X;
+						}
+					}
 				}
 			}
 
@@ -1982,7 +_,7 @@
 			Vector2 vector2 = default(Vector2);
 			for (int i = num; i < num2; i++) {
 				for (int j = num3; j < num4; j++) {
-					if (Main.tile[i, j] == null || Main.tile[i, j].slope() != 0 || Main.tile[i, j].inActive() || !Main.tile[i, j].active() || (Main.tile[i, j].type != 32 && Main.tile[i, j].type != 37 && Main.tile[i, j].type != 48 && Main.tile[i, j].type != 232 && Main.tile[i, j].type != 53 && Main.tile[i, j].type != 57 && Main.tile[i, j].type != 58 && Main.tile[i, j].type != 69 && Main.tile[i, j].type != 76 && Main.tile[i, j].type != 112 && Main.tile[i, j].type != 116 && Main.tile[i, j].type != 123 && Main.tile[i, j].type != 224 && Main.tile[i, j].type != 234 && Main.tile[i, j].type != 352 && Main.tile[i, j].type != 484))
+					if (Main.tile[i, j] == null || Main.tile[i, j].slope() != 0 || Main.tile[i, j].inActive() || !Main.tile[i, j].active() || (Main.tile[i, j].type != 32 && Main.tile[i, j].type != 37 && Main.tile[i, j].type != 48 && Main.tile[i, j].type != 232 && Main.tile[i, j].type != 53 && Main.tile[i, j].type != 57 && Main.tile[i, j].type != 58 && Main.tile[i, j].type != 69 && Main.tile[i, j].type != 76 && Main.tile[i, j].type != 112 && Main.tile[i, j].type != 116 && Main.tile[i, j].type != 123 && Main.tile[i, j].type != 224 && Main.tile[i, j].type != 234 && Main.tile[i, j].type != 352 && Main.tile[i, j].type != 484)) // TODO: && Main.tile[i, j].type != 348
 						continue;
 
 					vector2.X = i * 16;
@@ -2040,10 +_,12 @@
 						if (!fireImmune && (type == 37 || type == 58 || type == 76))
 							num5 = 20;
 
-						if (type == 48)
+						if (type == 48) // TODO: !Main.player[Main.myPlayer].trapImmune && !Main.player[Main.myPlayer].HasItemInArmor(3350) && !Main.player[Main.myPlayer].HasItemInArmor(3352) && !Main.player[Main.myPlayer].HasItemInArmor(3353) && !Main.player[Main.myPlayer].HasItemInArmor(3354) &&
 							num5 = 60;
 
-						if (type == 232)
+						// TODO: if (!Main.player[Main.myPlayer].trapImmune && !Main.player[Main.myPlayer].HasItemInArmor(3350) && !Main.player[Main.myPlayer].HasItemInArmor(3352) && !Main.player[Main.myPlayer].HasItemInArmor(3353) && !Main.player[Main.myPlayer].HasItemInArmor(3354) && type == 348) { num5 = 35; Main.player }
+
+						if (type == 232) // !Main.player[Main.myPlayer].trapImmune && !Main.player[Main.myPlayer].HasItemInArmor(3350) && !Main.player[Main.myPlayer].HasItemInArmor(3352) && !Main.player[Main.myPlayer].HasItemInArmor(3353) && !Main.player[Main.myPlayer].HasItemInArmor(3354) &&
 							num5 = 80;
 
 						if (type == 484)
