--- src/Terraria/Terraria/Item.cs
+++ src/Avalon/Terraria/Item.cs
@@ -39,8 +_,8 @@
 		public static int[] headType = new int[266];
 		public static int[] bodyType = new int[235];
 		public static int[] legType = new int[218];
-		public static bool[] staff = new bool[5045];
+		public static bool[] staff = new bool[Main.maxItemTypes];
-		public static bool[] claw = new bool[5045];
+		public static bool[] claw = new bool[Main.maxItemTypes];
 		public bool flame;
 		public bool mech;
 		public int noGrabDelay;
@@ -168,14 +_,7 @@
 			}
 		}
 
-		public bool FitsAccessoryVanitySlot {
-			get {
-				if (!canBePlacedInVanityRegardlessOfConditions && handOnSlot == -1 && handOffSlot == -1 && backSlot == -1 && frontSlot == -1 && shoeSlot == -1 && waistSlot == -1 && wingSlot == -1 && shieldSlot == -1 && neckSlot == -1 && faceSlot == -1)
-					return balloonSlot != -1;
-
-				return true;
-			}
-		}
+		public bool FitsAccessoryVanitySlot => true;
 
 		public int OriginalRarity => ContentSamples.ItemsByType[type].rare;
 
@@ -1077,8 +_,8 @@
 				if (rare < -1)
 					rare = -1;
 
-				if (rare > 11)
+				if (rare > 12)
-					rare = 11;
+					rare = 12;
 			}
 
 			num14 *= num14;
@@ -44361,7 +_,7 @@
 				playerIndexTheItemIsReservedFor = Main.myPlayer;
 
 			ResetStats(Type);
-			if (type >= 5045)
+			if (type >= Main.maxItemTypes)
 				type = 0;
 
 			if (type == 0) {
@@ -44540,7 +_,7 @@
 				material = ItemID.Sets.IsAMaterial[type];
 
 			RebuildTooltip();
-			if (type > 0 && type < 5045 && ItemID.Sets.Deprecated[type]) {
+			if (type > 0 && type < Main.maxItemTypes && ItemID.Sets.Deprecated[type]) {
 				netID = 0;
 				type = 0;
 				stack = 0;
@@ -45840,7 +_,7 @@
 				Prefix(reader.ReadByte());
 			}
 
-			if (type >= 5045)
+			if (type >= Main.maxItemTypes)
 				TurnToAir();
 		}
 
