--- src/Terraria/Terraria/Lang.cs
+++ src/Avalon/Terraria/Lang.cs
@@ -28,7 +_,7 @@
 		[Obsolete("Lang arrays have been replaced with the new Language.GetText system.")]
 		public static LocalizedText[] mp = new LocalizedText[23];
 		[Obsolete("Lang arrays have been replaced with the new Language.GetText system.")]
-		public static LocalizedText[] chestType = new LocalizedText[52];
+		public static LocalizedText[] chestType = new LocalizedText[Main.maxChests];
 		[Obsolete("Lang arrays have been replaced with the new Language.GetText system.")]
 		public static LocalizedText[] dresserType = new LocalizedText[40];
 		[Obsolete("Lang arrays have been replaced with the new Language.GetText system.")]
@@ -39,8 +_,8 @@
 		private static LocalizedText[] _projectileNameCache = new LocalizedText[950];
 		private static LocalizedText[] _npcNameCache = new LocalizedText[663];
 		private static LocalizedText[] _negativeNpcNameCache = new LocalizedText[65];
-		private static LocalizedText[] _buffNameCache = new LocalizedText[323];
+		private static LocalizedText[] _buffNameCache = new LocalizedText[Main.maxBuffTypes];
-		private static LocalizedText[] _buffDescriptionCache = new LocalizedText[323];
+		private static LocalizedText[] _buffDescriptionCache = new LocalizedText[Main.maxBuffTypes];
 		private static ItemTooltip[] _itemTooltipCache = new ItemTooltip[5045];
 		private static LocalizedText[] _emojiNameCache = new LocalizedText[145];
 
