--- src/Terraria/Terraria/Main.cs
+++ src/Avalon/Terraria/Main.cs
@@ -298,7 +_,7 @@
 		public const int maxNPCTypes = 663;
 		public const int maxTileSets = 623;
 		public const int maxWallTypes = 316;
-		public const int maxBuffTypes = 323;
+		public const int maxBuffTypes = BuffID.Count;
 		public const int maxGlowMasks = 301;
 		public const int maxExtras = 212;
 		public const int maxGoreTypes = 1269;
@@ -518,15 +_,15 @@
 		public static float musicPitch = 0f;
 		public static bool[] projHostile = new bool[950];
 		public static bool[] projHook = new bool[950];
-		public static bool[] pvpBuff = new bool[323];
+		public static bool[] pvpBuff = new bool[maxBuffTypes];
-		public static bool[] persistentBuff = new bool[323];
+		public static bool[] persistentBuff = new bool[maxBuffTypes];
-		public static bool[] vanityPet = new bool[323];
+		public static bool[] vanityPet = new bool[maxBuffTypes];
-		public static bool[] lightPet = new bool[323];
+		public static bool[] lightPet = new bool[maxBuffTypes];
-		public static bool[] meleeBuff = new bool[323];
+		public static bool[] meleeBuff = new bool[maxBuffTypes];
-		public static bool[] debuff = new bool[323];
+		public static bool[] debuff = new bool[maxBuffTypes];
-		public static bool[] buffNoSave = new bool[323];
+		public static bool[] buffNoSave = new bool[maxBuffTypes];
-		public static bool[] buffNoTimeDisplay = new bool[323];
+		public static bool[] buffNoTimeDisplay = new bool[maxBuffTypes];
-		public static bool[] buffDoubleApply = new bool[323];
+		public static bool[] buffDoubleApply = new bool[maxBuffTypes];
 		public static int maxMP = 10;
 		public static string[] recentWorld = new string[maxMP];
 		public static string[] recentIP = new string[maxMP];
@@ -636,7 +_,7 @@
 		public static bool armorHide;
 		public static float craftingAlpha = 1f;
 		public static float armorAlpha = 1f;
-		public static float[] buffAlpha = new float[323];
+		public static float[] buffAlpha = new float[maxBuffTypes];
 		public static bool hardMode;
 		public float chestLootScale = 1f;
 		public bool chestLootHover;
@@ -2177,24 +_,14 @@
 			}
 		}
 
-		public static bool masterMode {
-			get {
-				if (ActiveWorldFileData == null)
-					return false;
-
-				if (_overrideForMasterMode.HasValue)
-					return _overrideForMasterMode.Value;
-
-				return _currentGameModeInfo.IsMasterMode;
-			}
-		}
+		public static bool masterMode => false;
 
 		public static bool expertMode {
 			get {
 				if (ActiveWorldFileData == null)
 					return false;
 
-				if (_overrideForExpertMode.HasValue)
+				if (_overrideForExpertMode.HasValue || _overrideForMasterMode.HasValue)
 					return _overrideForExpertMode.Value;
 
 				return _currentGameModeInfo.IsExpertMode;
@@ -4725,7 +_,7 @@
 #if CLIENT
 			Configuration.Load();
 			graphics = new GraphicsDeviceManager(this as Game);
-			base.Content.RootDirectory = "Content";
+			base.Content.RootDirectory = "Avalon Content";
 #endif
 		}
 
@@ -8341,9 +_,9 @@
 
 		private void LoadMusic() {
 			if (SoundEngine.IsAudioSupported) {
-				engine = new AudioEngine("Content\\TerrariaMusic.xgs");
+				engine = new AudioEngine("Avalon Content\\TerrariaMusic.xgs");
-				soundBank = new SoundBank(engine, "Content\\Sound Bank.xsb");
+				soundBank = new SoundBank(engine, "Avalon Content\\Sound Bank.xsb");
-				waveBank = new WaveBank(engine, "Content\\Wave Bank.xwb");
+				waveBank = new WaveBank(engine, "Avalon Content\\Wave Bank.xwb");
 				for (int i = 1; i < 89; i++) {
 					music[i] = soundBank.GetCue("Music_" + i);
 				}
@@ -8351,9 +_,9 @@
 		}
 
 		private IEnumerator LoadMusic_InSteps() {
-			engine = new AudioEngine("Content\\TerrariaMusic.xgs");
+			engine = new AudioEngine("Avalon Content\\TerrariaMusic.xgs");
-			soundBank = new SoundBank(engine, "Content\\Sound Bank.xsb");
+			soundBank = new SoundBank(engine, "Avalon Content\\Sound Bank.xsb");
-			waveBank = new WaveBank(engine, "Content\\Wave Bank.xwb");
+			waveBank = new WaveBank(engine, "Avalon Content\\Wave Bank.xwb");
 			yield return null;
 			for (int i = 1; i < 89; i++) {
 				music[i] = soundBank.GetCue("Music_" + i);
@@ -38852,10 +_,16 @@
 					if (num107 == 3)
 						num109 = 2;
 
-					Vector2 origin2 = FontAssets.MouseText.Value.MeasureString(versionNumber);
+					string avalonVersion = "Exxo Avalon: Origins v1.0.0.0";
+
+					Vector2 origin2 = FontAssets.MouseText.Value.MeasureString("Terraria " + versionNumber);
 					origin2.X *= 0.5f;
+					spriteBatch.DrawString(FontAssets.MouseText.Value, "Terraria " + versionNumber, new Vector2(origin2.X + (float)num108 + 10f, (float)screenHeight - origin2.Y + (float)num109 - 2f), color12, 0f, origin2, 1f, SpriteEffects.None, 0f);
+
+					Vector2 origin3 = FontAssets.MouseText.Value.MeasureString(avalonVersion);
+					origin3.X *= 0.5f;
-					origin2.Y *= 0.5f;
+					origin3.Y *= 0.5f;
-					spriteBatch.DrawString(FontAssets.MouseText.Value, versionNumber, new Vector2(origin2.X + (float)num108 + 10f, (float)screenHeight - origin2.Y + (float)num109 - 2f), color12, 0f, origin2, 1f, SpriteEffects.None, 0f);
+					spriteBatch.DrawString(FontAssets.MouseText.Value, avalonVersion, new Vector2(origin3.X + (float)num108 + 10f, (float)screenHeight - origin3.Y + (float)num109 - 2f), color12, 0f, origin3, 1f, SpriteEffects.None, 0f);
 				}
 			}
 
