--- src/Terraria/Terraria/MessageBuffer.cs
+++ src/Avalon/Terraria/MessageBuffer.cs
@@ -546,6 +_,8 @@
 						NPC.downedEmpressOfLight = bitsByte22[5];
 						NPC.downedQueenSlime = bitsByte22[6];
 						Main.getGoodWorld = bitsByte22[7];
+						BitsByte bitsByte23 = reader.ReadByte();
+						NPC.downedGolemSinceCultistsKilled = bitsByte23[0];
 						WorldGen.SavedOreTiers.Copper = reader.ReadInt16();
 						WorldGen.SavedOreTiers.Iron = reader.ReadInt16();
 						WorldGen.SavedOreTiers.Silver = reader.ReadInt16();
@@ -1141,7 +_,7 @@
 							Item item2 = Main.item[num72];
 							ItemSyncPersistentStats itemSyncPersistentStats = default(ItemSyncPersistentStats);
 							itemSyncPersistentStats.CopyFrom(item2);
-							bool newAndShiny = (item2.newAndShiny || item2.netID != num71) && ItemSlot.Options.HighlightNewItems && (num71 < 0 || num71 >= 5045 || !ItemID.Sets.NeverAppearsAsNewInInventory[num71]);
+							bool newAndShiny = (item2.newAndShiny || item2.netID != num71) && ItemSlot.Options.HighlightNewItems && (num71 < 0 || num71 >= Main.maxItemTypes || !ItemID.Sets.NeverAppearsAsNewInInventory[num71]);
 							item2.netDefaults(num71);
 							item2.newAndShiny = newAndShiny;
 							item2.Prefix(pre2);
