--- src/Terraria/Terraria/NetMessage.cs
+++ src/Avalon/Terraria/NetMessage.cs
@@ -290,6 +_,9 @@
 							bb13[6] = NPC.downedQueenSlime;
 							bb13[7] = Main.getGoodWorld;
 							writer.Write(bb13);
+							BitsByte bb14 = 0;
+							bb14[0] = NPC.downedGolemSinceCultistsKilled;
+							writer.Write(bb14);
 							writer.Write((short)WorldGen.SavedOreTiers.Copper);
 							writer.Write((short)WorldGen.SavedOreTiers.Iron);
 							writer.Write((short)WorldGen.SavedOreTiers.Silver);
