--- src/Terraria/Terraria/NPC.cs
+++ src/Avalon/Terraria/NPC.cs
@@ -24,7 +_,7 @@
 
 namespace Terraria
 {
-	public class NPC : Entity
+	public partial class NPC : Entity
 	{
 		private const int NPC_TARGETS_START = 300;
 		public bool IsABestiaryIconDummy;
@@ -110,7 +_,7 @@
 		public const int maxBuffs = 5;
 		public int[] buffType = new int[5];
 		public int[] buffTime = new int[5];
-		public bool[] buffImmune = new bool[323];
+		public bool[] buffImmune = new bool[Main.maxBuffTypes];
 		public bool midas;
 		public bool ichor;
 		public bool onFire;
@@ -2837,7 +_,7 @@
 				buffType[k] = 0;
 			}
 
-			for (int l = 0; l < 323; l++) {
+			for (int l = 0; l < Main.maxBuffTypes; l++) {
 				buffImmune[l] = false;
 			}
 
@@ -10878,7 +_,7 @@
 
 				noGravity = true;
 				rarity = 2;
-				for (int num2 = 0; num2 < 323; num2++) {
+				for (int num2 = 0; num2 < Main.maxBuffTypes; num2++) {
 					buffImmune[num2] = true;
 				}
 			}
@@ -11612,7 +_,7 @@
 			}
 
 			if (flag) {
-				for (int num4 = 0; num4 < 323; num4++) {
+				for (int num4 = 0; num4 < Main.maxBuffTypes; num4++) {
 					buffImmune[num4] = true;
 				}
 			}
@@ -56648,7 +_,7 @@
 
 			if (type == 1 && ai[1] > 0f) {
 				int num3 = (int)ai[1];
-				if (num3 > 0 && num3 < 5045) {
+				if (num3 > 0 && num3 < Main.maxItemTypes) {
 					int stackForSlimeItemDrop = GetStackForSlimeItemDrop(num3);
 					Item.NewItem((int)position.X, (int)position.Y, width, height, num3, stackForSlimeItemDrop);
 				}
@@ -57383,6 +_,7 @@
 				}
 
 				SetEventFlagCleared(ref downedGolemBoss, 6);
+                downedGolemSinceCultistsKilled = true;
 			}
 
 			if (type == 471 && (Main.expertMode || player.RollLuck(2) == 0)) {
@@ -58500,6 +_,7 @@
 
 				Item.NewItem((int)position.X, (int)position.Y, width, height, 3549, 1, noBroadcast: false, -1);
 				WorldGen.TriggerLunarApocalypse();
+                downedGolemSinceCultistsKilled = false;
 			}
 
 			if (type == 398) {
@@ -59495,6 +_,7 @@
 					break;
 				case 245:
 					SetEventFlagCleared(ref downedGolemBoss, 6);
+					downedGolemSinceCultistsKilled = true;
 					break;
 				case 370:
 					SetEventFlagCleared(ref downedFishron, 7);
@@ -59532,6 +_,7 @@
 				case 439:
 					SetEventFlagCleared(ref downedAncientCultist, 9);
 					WorldGen.TriggerLunarApocalypse();
+					downedGolemSinceCultistsKilled = false;
 					break;
 				case 398:
 					SetEventFlagCleared(ref downedMoonlord, 10);
@@ -71263,7 +_,7 @@
 		}
 
 		public void RequestBuffRemoval(int buffTypeToRemove) {
-			if (buffTypeToRemove < 0 || buffTypeToRemove >= 323 || !BuffID.Sets.CanBeRemovedByNetMessage[buffTypeToRemove])
+			if (buffTypeToRemove < 0 || buffTypeToRemove >= Main.maxBuffTypes || !BuffID.Sets.CanBeRemovedByNetMessage[buffTypeToRemove])
 				return;
 
 			int num = FindBuffIndex(buffTypeToRemove);
