--- src/Terraria/Terraria/Player.cs
+++ src/Avalon/Terraria/Player.cs
@@ -686,7 +_,7 @@
 		public const int maxBuffs = 22;
 		public int[] buffType = new int[22];
 		public int[] buffTime = new int[22];
-		public bool[] buffImmune = new bool[323];
+		public bool[] buffImmune = new bool[Main.maxBuffTypes];
 		public int heldProj = -1;
 		public int breathCD;
 		public int breathMax = 200;
@@ -17379,7 +_,7 @@
 					AddBuff(194, 2, quiet: false);
 			}
 
-			for (int num26 = 0; num26 < 323; num26++) {
+			for (int num26 = 0; num26 < Main.maxBuffTypes; num26++) {
 				buffImmune[num26] = false;
 			}
 
@@ -25720,6 +_,7 @@
 		}
 
 		public void Spawn(PlayerSpawnContext context) {
+			Main.InitNPCLifeBytes();
 			Main.LocalPlayer.creativeInterface = false;
 			_funkytownAchievementCheckCooldown = 100;
 			bool flag = false;
@@ -36982,7 +_,7 @@
 								if (num >= 58) {
 									for (int num7 = 0; num7 < 58; num7++) {
 										int num8 = binaryReader.ReadInt32();
-										if (num8 >= 5045) {
+										if (num8 >= Main.maxItemTypes) {
 											player.inventory[num7].netDefaults(0);
 											binaryReader.ReadInt32();
 											binaryReader.ReadByte();
@@ -37001,7 +_,7 @@
 								else {
 									for (int num9 = 0; num9 < 48; num9++) {
 										int num10 = binaryReader.ReadInt32();
-										if (num10 >= 5045) {
+										if (num10 >= Main.maxItemTypes) {
 											player.inventory[num9].netDefaults(0);
 											binaryReader.ReadInt32();
 											binaryReader.ReadByte();
@@ -37019,7 +_,7 @@
 										for (int num11 = 0; num11 < 5; num11++) {
 											if (num11 != 1) {
 												int num12 = binaryReader.ReadInt32();
-												if (num12 >= 5045) {
+												if (num12 >= Main.maxItemTypes) {
 													player.miscEquips[num11].netDefaults(0);
 													binaryReader.ReadByte();
 												}
@@ -37029,7 +_,7 @@
 												}
 
 												num12 = binaryReader.ReadInt32();
-												if (num12 >= 5045) {
+												if (num12 >= Main.maxItemTypes) {
 													player.miscDyes[num11].netDefaults(0);
 													binaryReader.ReadByte();
 												}
@@ -37043,7 +_,7 @@
 									else {
 										for (int num13 = 0; num13 < 5; num13++) {
 											int num14 = binaryReader.ReadInt32();
-											if (num14 >= 5045) {
+											if (num14 >= Main.maxItemTypes) {
 												player.miscEquips[num13].netDefaults(0);
 												binaryReader.ReadByte();
 											}
@@ -37053,7 +_,7 @@
 											}
 
 											num14 = binaryReader.ReadInt32();
-											if (num14 >= 5045) {
+											if (num14 >= Main.maxItemTypes) {
 												player.miscDyes[num13].netDefaults(0);
 												binaryReader.ReadByte();
 											}
