--- src/Terraria/Terraria/Terraria.csproj
+++ src/Avalon/Terraria/Terraria.csproj
@@ -63,8 +_,8 @@
     <EmbeddedResource Remove="Libraries/Mac/**" Condition="!$(DefineConstants.Contains('MAC'))" />
     <EmbeddedResource Remove="Libraries/Linux/**" Condition="!$(DefineConstants.Contains('LINUX'))" />
     <EmbeddedResource Remove="Libraries/Mono/**" Condition="!$(DefineConstants.Contains('MONO'))" />
-    <EmbeddedResource Remove="Libraries/XNA/**"/>
+    <EmbeddedResource Remove="Libraries/XNA/**" />
-    <EmbeddedResource Remove="Libraries/FNA/**"/>
+    <EmbeddedResource Remove="Libraries/FNA/**" />
     <EmbeddedResource Include="Localization/Content/**" />
     <EmbeddedResource Include="Microsoft/**" />
   </ItemGroup>
