--- src/Terraria/Terraria/GameContent/TextureAssets.cs
+++ src/Avalon/Terraria/GameContent/TextureAssets.cs
@@ -177,7 +_,7 @@
 		public static Asset<Texture2D>[] NpcToggle = new Asset<Texture2D>[2];
 		public static Asset<Texture2D>[] HbLock = new Asset<Texture2D>[2];
 		public static Asset<Texture2D>[] blockReplaceIcon = new Asset<Texture2D>[2];
-		public static Asset<Texture2D>[] Buff = new Asset<Texture2D>[323];
+		public static Asset<Texture2D>[] Buff = new Asset<Texture2D>[Main.maxBuffTypes];
 		public static Asset<Texture2D>[] Item = new Asset<Texture2D>[5045];
 		public static Asset<Texture2D>[] ItemFlame = new Asset<Texture2D>[5045];
 		public static Asset<Texture2D>[] Npc = new Asset<Texture2D>[663];
