--- src/Terraria/Terraria/GameContent/TownRoomManager.cs
+++ src/Avalon/Terraria/GameContent/TownRoomManager.cs
@@ -10,7 +_,7 @@
 	{
 		public static object EntityCreationLock = new object();
 		private List<Tuple<int, Point>> _roomLocationPairs = new List<Tuple<int, Point>>();
-		private bool[] _hasRoom = new bool[665];
+		private bool[] _hasRoom = new bool[Main.maxNPCTypes];
 
 		public void AddOccupantsToList(int x, int y, List<int> occupantsList) {
 			AddOccupantsToList(new Point(x, y), occupantsList);
